src/libs/NextStepUtils.ts

```diff
@@ -27,5 +50,270 @@ function parseMessage(messages: Message[] | undefined) {
     return `<next-step>${formattedHtml}</next-step>`;
 }
 
-// eslint-disable-next-line import/prefer-default-export
-export {parseMessage};
+type BuildNextStepParameters = {
+    isPaidWithWallet?: boolean;
+};
+
+/**
+ * Generates an optimistic nextStep based on a current report status and other properties.
+ *
+ * @param report
+ * @param predictedNextStatus - a next expected status of the report
+ * @param parameters.isPaidWithWallet - Whether a report has been paid with the wallet or outside of Expensify
+ * @returns nextStep
+ */
+function buildNextStep(report: Report | EmptyObject, predictedNextStatus: ValueOf<typeof CONST.REPORT.STATUS_NUM>, {isPaidWithWallet}: BuildNextStepParameters = {}): ReportNextStep | null {
+    const {policyID = '', ownerAccountID = -1, managerID = -1} = report;
```

Comment:
Can we return early if the passed report is not Expense report?

Reply:
Done - 12a28c1.